tests/gitlab/generate_spack_upstream.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# This script is meant be launch in the root directory of MFEM do compile all
+# the dependencies of MFEM for all the specs in CI. By doing so, we obtain a
+# Spack instance that can be used as an upstream.
+
+# This patch applies shared permissions to Spack installations.
+git apply tests/gitlab/upstream-permission.patch
+
+# call uberenv for all specs in CI
+git grep -e "^[^#]" .gitlab | grep "SPEC" \
```

Comment:
make sure this works on both Mac and Linux

Reply:
Done.